localstack/services/kms/provider.py

```diff
@@ -133,89 +163,238 @@ def __init__(self, message=None):
+    def replicate_key(
+        self, context: RequestContext, request: ReplicateKeyRequest
+    ) -> ReplicateKeyResponse:
+        replicate_from_store = self._get_store(context)
+        key = replicate_from_store.get_key(request.get("KeyId"))
+        key_id = key.metadata.get("KeyId")
+        if not key.metadata.get("MultiRegion"):
+            raise UnsupportedOperationException(
+                f"Unable to replicate a non-MultiRegion key {key_id}"
+            )
+        replica_region = request.get("ReplicaRegion")
+        replicate_to_store = kms_stores[context.account_id][replica_region]
+        if key_id in replicate_to_store.keys:
+            raise AlreadyExistsException(
+                f"Unable to replicate key {key_id} to region {replica_region}, as the key "
+                f"already exist there"
+            )
+        replica_key = copy.deepcopy(key)
+        if request.get("Description"):
+            replica_key.metadata["Description"] = request.get("Description")
+        # Multiregion keys have the same key ID for all replicas, but ARNs differ, as they include actual regions of
+        # replicas.
+        replica_key.calculate_and_set_arn(context.account_id, replica_region)
+        replicate_to_store.keys[key_id] = replica_key
+
+    @handler("UpdateKeyDescription", expand=False)
+    def update_key_description(
+        self, context: RequestContext, request: UpdateKeyDescriptionRequest
+    ) -> None:
+        key = self._get_store(context).get_key(request.get("KeyId"))
+        key.metadata["Description"] = request.get("Description")
 
+    # TODO If there is an attempt to create a grant for a key pending deletion, the following error is generated in AWS:
+    #      KMSInvalidStateException("<KEY ARN> is pending deletion")
+    #  Should implement that in LocalStack as well.
     @handler("CreateGrant", expand=False)
     def create_grant(
-        self, context: RequestContext, create_grant_request: CreateGrantRequest
+        self, context: RequestContext, request: CreateGrantRequest
     ) -> CreateGrantResponse:
-        self._validate_grant(create_grant_request)
-        region_details = KMSBackend.get()
-
-        grant = dict(create_grant_request)
-        grant[GRANT_ID] = long_uid()
-        grant[GRANT_TOKENS] = [long_uid()]
-        if NAME not in grant:
-            grant[NAME] = ""
-        grant[CREATION_DATE] = time.time()
-
-        region_details.grants[grant[GRANT_ID]] = grant
-        return CreateGrantResponse(GrantId=grant[GRANT_ID], GrantToken=grant[GRANT_TOKENS][0])
+        store = self._get_store(context)
+        # KeyId can potentially hold one of multiple different types of key identifiers. get_key finds a key no
+        # matter which type of id is used.
+        request["KeyId"] = store.get_canonical_key_id(request.get("KeyId"))
+        self._validate_grant_request(request, store)
+        grant_name = request.get("Name")
+        grant = None
```

Comment:
No need for the `None` assignment here as all the following execution paths define and assign this variable either way.

Reply:
It is more like a C++ habit. :-) Thanks.